### ISSUE TITLE
SwaySwap Indexer Integration Checklist

### ISSUE THREAD

**Issue by ra0x3** [FuelLabs/fuel-indexer#147]
A list of issues and PRs that are either required or (very) nice to have for a successful SwaySwap/Indexer integration
### Required
- [x] Node configuration
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/47
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/33
- [x] Docker support
  - [x] https://github.com/FuelLabs/fuel-indexer/pull/116
- [x] DNS
  - [x] Acquire SwaySwap indexer hostname (http://index.swayswap.io/)
- [x] Deployment config
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/145
  - [x] https://github.com/FuelLabs/fuel-deployment/issues/76
- [x] Public Postgres resources
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/146
- [x] Create schema for types/structs/entities
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/149
- [x] Support indexing blocks and transactions
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/151
  - [x] https://github.com/FuelLabs/fuel-indexer/pull/178
- [x] Sqlite support
  - [x] https://github.com/FuelLabs/fuel-indexer/pull/134
- [x] End-to-end test
  - [x] https://github.com/FuelLabs/fuel-indexer/pull/143
- [x] Foreign Key Support
  - [x] https://github.com/FuelLabs/fuel-indexer/issues/112

### Nice to have
- [x] *Schema upload and versioning
  - https://github.com/FuelLabs/fuel-indexer/issues/138
- [x] *WASM upload and versioning
  - https://github.com/FuelLabs/fuel-indexer/issues/137
- [x] Release CI/CD 
  - Where changes get at least pushed to a test environment
- ~~Ensure all [built-in types](https://fuellabs.github.io/sway/v0.19.2/basics/built_in_types.html) and [structs using built-in types](https://fuellabs.github.io/sway/v0.19.2/basics/structs_tuples_and_enums.html) can be indexed~~
  - Whether while passing a single `Event` to a `handler`, or passing multiple
  - Might need to be done before a public-facing release
- ~~Query model serialization~~
  - ~~https://github.com/FuelLabs/fuel-indexer/issues/136~~
 
_**Would allow for making index changes without a new release or downtime - and also would allow for multiple apps per service*_

**Comment by ra0x3** [comment]
@JoshuaBatty @tjsharp1 @mitchmindtree I added this as way to sort've transparently keep track of what needs to be done for SwaySwap. Feel free to add more PRs/issues if I've left anything out.
